### ISSUE TITLE
Implement data source for azurerm_automation_dsc_configuration_metadata

### ISSUE THREAD

**Issue by bastienfrancheteau** [terraform-providers/terraform-provider-azurerm#2162]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

When registering a VM as a DSC node, the Automation Account's key and DSC Registration URL values are required. These cannot be retrieved in Terraform. Creating an azurerm_automation_dsc_configuration_metadata module that returns these 2 values would enable us to dynamically register Azure VMs as DSC nodes.

Suggested name for new data source: azurerm_automation_dsc_configuration_metadata

* azurerm_automation_dsc_configuration

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by whytoe** [comment]
The resource already supports outputting the Primary Key/DSC Server Endpoint
https://www.terraform.io/docs/providers/azurerm/r/automation_account.html

Is there a blocker for this? @tombuildsstuff

**Comment by whytoe** [comment]
@tombuildsstuff does the GoSDK support this now, its been almost a year

It is available in powershell
```
PS C:\WINDOWS\system32> $data = Get-AzAutomationRegistrationInfo -ResourceGroupName resourcegroup -AutomationAccountName AutomationAccount

ResourceGroupName     : resourcegroup
AutomationAccountName : AutomationAccount
PrimaryKey            : PrimaryKey            
SecondaryKey          : SecondaryKey
Endpoint              : https://cc-agentservice-prod-1.azure-automation.net/accounts/guid
```

**Comment by mbfrahry** [comment]
It looks like this issue is resolved through the #4740  that @whytoe created. That should go out in 1.37.0 of the provider. Feel free to create a new issue if more work needs to be done here.
